src/TableView/TableView.jsx

```diff
@@ -398,7 +447,9 @@ export const getTableView = (schema, Table = TableComponent, isChildView = false
             onChange: this.handleFilterData,
             by: filterBy,
             value: filterValue,
-            onlyStringTypes: true
+            onlyStringTypes: true,
+            includeRelations: !substringSearchEnabled || metadataSubstringSearchEnabled === false,
```

Comment:
Please use consistent syntax when `substringSearchEnabled` and `metadataSubstringSearchEnabled` are used with logical operators. They are both boolean values so it's not necessary to compare them with boolean value.
```suggestion
            includeRelations: !substringSearchEnabled || !metadataSubstringSearchEnabled,
```